crates/node/engine/src/task_queue/tasks/forkchoice/task.rs

```diff
@@ -1,36 +1,90 @@
 //! A task for the `engine_forkchoiceUpdated` method, with no attributes.
 
 use crate::{
-    EngineClient, EngineState, EngineTaskError, EngineTaskExt, ForkchoiceTaskError, Metrics,
+    EngineClient, EngineForkchoiceVersion, EngineState, EngineTaskError, EngineTaskExt,
+    ForkchoiceTaskError, Metrics,
 };
-use alloy_rpc_types_engine::INVALID_FORK_CHOICE_STATE_ERROR;
+use alloy_provider::ext::EngineApi;
+use alloy_rpc_types_engine::{INVALID_FORK_CHOICE_STATE_ERROR, PayloadId, PayloadStatusEnum};
 use async_trait::async_trait;
+use kona_genesis::RollupConfig;
+use kona_protocol::OpAttributesWithParent;
 use op_alloy_provider::ext::engine::OpEngineApi;
 use std::sync::Arc;
+use tokio::time::Instant;
 
 /// The [`ForkchoiceTask`] executes an `engine_forkchoiceUpdated` call with the current
 /// [`EngineState`]'s forkchoice, and no payload attributes.
 #[derive(Debug, Clone)]
 pub struct ForkchoiceTask {
     /// The engine client.
     pub client: Arc<EngineClient>,
+    /// The rollup config.
+    pub rollup: Arc<RollupConfig>,
+    /// Optional payload attributes to be used for the forkchoice update.
+    envelope: Option<OpAttributesWithParent>,
 }
 
 impl ForkchoiceTask {
     /// Creates a new [`ForkchoiceTask`].
-    pub const fn new(client: Arc<EngineClient>) -> Self {
-        Self { client }
+    pub const fn new(
+        client: Arc<EngineClient>,
+        rollup: Arc<RollupConfig>,
+        payload_attributes: Option<OpAttributesWithParent>,
+    ) -> Self {
+        Self { client, rollup, envelope: payload_attributes }
+    }
+
+    /// Checks the response of the `engine_forkchoiceUpdated` call, and updates the sync status if
+    /// necessary.
+    fn check_forkchoice_updated_status(
```

Comment:
This check is strictly different than the checks we performed on synchronizing the local engine state beforehand. i.e., we'd never return a temporary error if the engine was syncing.